### ISSUE TITLE
Ram usage above the set limit.

### ISSUE THREAD

**Issue by polhaghverdian** [cockpit-project/cockpit-machines#99]
Cockpit version: 241
OS: Ubuntu 20.04
Page: VM - overview

Ram usage is above the limit: **4.2/4**

![bild](https://user-images.githubusercontent.com/50968806/114297448-03e28d80-9ab1-11eb-9574-8bf10ebb5cff.png)

**Comment by KKoukiou** [comment]
HI @polhaghverdian can you please check the following command output and paste the results here.

```
$ sudo virsh domstats  --balloon  cirros

```
Replace cirros with your VM's name.

You should see something like:

```
Domain: 'cirros'
  balloon.current=2062336
  balloon.maximum=2062336
  balloon.swap_in=0
  balloon.last-update=1618303532
  balloon.rss=147728
```

In the progress bar you see in the UI, we use current for the upper limit and rss for the actual usage. So let's verify that the values presented are the ones libvirt gives us and then I can dig into that.

**Comment by KKoukiou** [comment]
anyway, I guess the values are probably same in the CLI, (will wait for your verification) but I have an idea about your issue already.

* The progress bar max value is the actual allocation of memory for the guest.
This value can be less than the maximum allocation, (can be set in the memory dialog) to allow for ballooning up the guests memory on the fly. In case maximum memory > current memory, when the VM boots, and before the ballon driver is loaded you can see memory usage above the current memory value, which as I said is the upper limit of the progress bar.

Is this the case?

**Comment by polhaghverdian** [comment]
@KKoukiou 

Here is another VM with same problem: **2.2/2**

![bild](https://user-images.githubusercontent.com/50968806/114600519-bc1c5b80-9c94-11eb-9b10-016a8add71b0.png)

Domain: 'VM2'
  balloon.current=2097152
  balloon.maximum=2097152
  balloon.swap_in=0
  balloon.swap_out=0
  balloon.major_fault=0
  balloon.minor_fault=0
  balloon.unused=1729172
  balloon.available=2024696
  balloon.usable=1660580
  balloon.last-update=1618258276
  balloon.disk_caches=271216
  balloon.hugetlb_pgalloc=0
  balloon.hugetlb_pgfail=0
  balloon.rss=2289252

**Comment by KKoukiou** [comment]
Ah I can explain @polhaghverdian. The memory progress bar shows:

balloon.rss/balloon.current <= in your case 2.2 / 2 GiB

RSS is the memory the qemu process is using in the host, not the amount of memory in use by the guest as seen by the guest.
However Balloon current is the maximum memory the guest can use as seen by the guest.

In other words, the qemu process uses more memory than what it makes available to the guest, and in this case, seemingly your guest is using all the available memory they have, which results in this confusing progress bar.

So esentially we should either use some other value for memory, the memory seen by the guest or document what these values are. 

For this we have an open issue https://github.com/cockpit-project/cockpit-machines/issues/49.

So I will close this, since it's essentially a duplicate of the issue above. Thanks for taking the time to report this issue :)
